File: Daily Flash/week 4/day 1/18-DailyFlash_Solutions/3_Feb_Solutions_One/Java/Program2.java
import java.io.*;


class Program2{

	public static void main(String args[])throws IOException{


		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int rem,position=1,power=1;
		System.out.printf("Enter four digit the binary number \n");
		int num = Integer.parseInt(br.readLine());
		int sum = 0;

		while(num!=0){

			rem = num%10;

			if(rem == 1){

				sum = sum + power;
			}

			power = powerOf2(2,position);
			position++;

			num = num/10;

		}

		switch(sum){

			case 10:
				System.out.printf("A\n");
				break;
			case 11:
				System.out.printf("B\n");
				break;
			case 12:
				System.out.printf("C\n");
				break;
			case 13:
				System.out.printf("D\n");
				break;
			case 14:
				System.out.printf("E\n");
				break;
			case 15:
				System.out.printf("F\n");
				break;
			default:
				System.out.printf("%d\n",sum);
		}


		

	}
	
	
	static int powerOf2(int under , int upper){


		while(upper !=1){
		
			under = 2*under;
			upper--;
		}	

		return under;

	}		
}	

File: Daily Flash/week 4/day 1/18-DailyFlash_Solutions/3_Feb_Solutions_One/Python/program2.py


num  = input("Enter the four digit binary number : ")

rem = 0
position = 1
power = 1
sum1 = 0


def powerOf2(under,upper):
    while(upper!=1):
        under = 2*under
        upper = upper-1
    return under    


while num!=0:

    rem = num%10;

    if rem==1:
        sum1 = sum1 + power;
    power = powerOf2(2,position)
    position = position + 1

    num = num/10

if sum1 == 10:
    print("A")
elif sum1 == 11:
    print("B")
elif sum1 == 12:
    print("C")
elif sum1 == 13:
    print("D")
elif sum1 == 14:
    print("E")
elif sum1 == 15:
    print("F")
else:
    print(sum1)

